clients/drcachesim/tools/syscall_mix.h

```diff
@@ -68,12 +68,24 @@ class syscall_mix_t : public analysis_tool_t {
     std::string
     parallel_shard_error(void *shard_data) override;
 
+    struct statistics_t {
+        std::unordered_map<int, int64_t> syscall_counts;
+        std::unordered_map<int, int64_t> syscall_trace_counts;
+        // Failure code counts, but only for system calls traced
+        // with -record_syscall where TRACE_MARKER_TYPE_SYSCALL_FAILED
+        // is recorded. That marker stores 32-bit errno values, which
+        // is our key here.
+        std::unordered_map<int, int64_t> syscall_errno_counts;
```

Comment:
I expected the failure counts to be split by syscall. Are they useful this way?

Reply:
For my purposes this is what I want: the count of EINTR ignoring the syscall number.  It may be likely that other use cases would want it split by syscall number though; and making me sum up across numbers might be worth the feature.  OK, I split by syscall number.